domain/network/state/package_test.go

```diff
@@ -76,3 +77,90 @@ func (s *linkLayerBaseSuite) checkRowCount(c *tc.C, table string, expected int)
 	c.Assert(err, tc.IsNil, tc.Commentf("counting rows in table %q", table))
 	c.Check(obtained, tc.Equals, expected, tc.Commentf("count of %q rows", table))
 }
+
+// addLinkLayerDevice adds a link layer device to the database and returns its UUID.
+func (s *linkLayerBaseSuite) addLinkLayerDevice(
+	c *tc.C, netNodeUUID, name, macAddress string,
+	deviceType corenetwork.LinkLayerDeviceType,
+) string {
+	deviceUUID := "device-" + name + "-uuid"
+
+	deviceTypeID, err := encodeDeviceType(deviceType)
+	c.Assert(err, tc.ErrorIsNil)
+
+	mtu := int64(1500)
+
+	s.query(c, `
+		INSERT INTO link_layer_device (uuid, net_node_uuid, name, mtu, 
+		                               mac_address, device_type_id, 
+		                               virtual_port_type_id, is_auto_start, 
+		                               is_enabled, is_default_gateway, gateway_address, vlan_tag)
+		                               	VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+	`, deviceUUID, netNodeUUID, name, mtu, macAddress, deviceTypeID, 0, true,
+		true, false, nil, 0)
+
+	return deviceUUID
+}
+
+func (s *linkLayerBaseSuite) addDNSDomains(c *tc.C, deviceUUID string, dnsDomains ...string) {
+	for _, dnsDomain := range dnsDomains {
+		s.query(c, `
+INSERT INTO link_layer_device_dns_domain (device_uuid, search_domain) 
+VALUES (?, ?)`, deviceUUID, dnsDomain)
+	}
+}
+
+func (s *linkLayerBaseSuite) addDNSAddresses(c *tc.C, deviceUUID string, dnsAddresses ...string) {
+	for _, dnsAddress := range dnsAddresses {
+		s.query(c, `
+INSERT INTO link_layer_device_dns_address (device_uuid, dns_address) 
+VALUES (?, ?)`, deviceUUID, dnsAddress)
+	}
+}
+
+func (s *linkLayerBaseSuite) addSubnet(c *tc.C, cidr string, spaceUUID corenetwork.SpaceUUID) string {
```

Comment:
todo: Make spaceUUID a string, future tests to be merged with `addSubnet` need to provide other UUIDs, it'll be easier.